pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -113,3 +127,43 @@ func (c *ClusterLoggingSyncer) createOrUpdateClusterConfigMap() error {
 	}
 	return utils.UpdateConfigMap(loggingconfig.ClusterConfigPath, loggingconfig.ClusterLoggingName, "cluster", c.configmaps)
 }
+
+func updateEmbeddedEndpoint(podLister v1.PodLister, serviceLister v1.ServiceLister, clusterLoggings v3.ClusterLoggingInterface) {
+	for {
+		select {
+		case <-timeout:
```

Comment:
Looks like the timeout is used for updating endpoint, but the variable is initialized as a package variable. You might want to initialize it in the beginning of the function.